patches/0096-Modify-strptime-source-to-work-with-uktest.patch

```diff
@@ -0,0 +1,124 @@
+@@ -33,21 +33,23 @@ static void checkStrptime(const char *s, const char *format, const struct tm *ex
+ 		strftime(buf1, sizeof(buf1), "%FT%H:%M:%S%Z", expected);
+ 		strftime(buf2, sizeof(buf2), "%FT%H:%M:%S%Z", &tm);
+ 
+-		t_error("\"%s\": for \"%s\" expected %s but got %s\n", format, s, buf1, buf2);
++		return -1;
+ 	}
++	return 0;
+ }
+ 
+-static void checkStrptimeTz(const char *s, int h, int m) {
++static int checkStrptimeTz(const char *s, int h, int m) {
+ 	long int expected = h * 3600 + m * 60;
+ 	struct tm tm = { };
+ 	const char *ret;
+ 
+ 	ret = strptime(s, "%z", &tm);
+ 	if (!ret || *ret != '\0') {
+-		t_error("\"%%z\": failed to parse \"%s\"\n", s);
++		return -1;
+ 	} else if (tm.tm_gmtoff != expected) {
+-		t_error("\"%%z\": for \"%s\" expected tm_gmtoff %ld but got %ld\n", s, tm.tm_gmtoff, expected);
++		return -1;
+ 	}
++	return 0;
+ }
+ 
+ static struct tm tm1 = {
+@@ -86,33 +88,33 @@ static struct tm tm4 = {
+ 	.tm_year = 1856 - 1900,
+ };
+ 
+-int main() {
++UK_TESTCASE(uk_libc_testsuite, strptime_tests)
++{
+ 	setenv("TZ", "UTC0", 1);
+ 
+ 	/* Time */
+-	checkStrptime("20:57:08", "%H:%M:%S", &tm1);
+-	checkStrptime("20:57:8", "%R:%S", &tm1);
+-	checkStrptime("20:57:08", "%T", &tm1);
++	UK_TEST_EXPECT_NOT_NULL("Parse time and check if it matches the expected value");
++	UK_TEST_EXPECT_ZERO(checkStrptime("20:57:08", "%H:%M:%S", &tm1));
++	UK_TEST_EXPECT_ZERO(checkStrptime("20:57:8", "%R:%S", &tm1));
++	UK_TEST_EXPECT_ZERO(checkStrptime("20:57:08", "%T", &tm1));
+ 
+ 	/* Format */
+-	checkStrptime("20:57:08", "%H : %M  :  %S", &tm1);
+-	checkStrptime("20 57  08", "%H %M %S", &tm1);
+-	checkStrptime("20%57%08", "%H %% %M%%%S", &tm1);
+-	checkStrptime("foo20bar57qux08      ", "foo %Hbar %M qux%S ", &tm1);
++	UK_TEST_EXPECT_ZERO(checkStrptime("20:57:08", "%H : %M  :  %S", &tm1));
++	UK_TEST_EXPECT_ZERO(checkStrptime("20 57  08", "%H %M %S", &tm1));
++	UK_TEST_EXPECT_ZERO(checkStrptime("20%57%08", "%H %% %M%%%S", &tm1));
++	UK_TEST_EXPECT_ZERO(checkStrptime("foo20bar57qux08      ", "foo %Hbar %M qux%S ", &tm1));
+ 
+ 	/* Date */
+-	checkStrptime("1991-08-25", "%Y-%m-%d", &tm2);
+-	checkStrptime("25.08.91", "%d.%m.%y", &tm2);
+-	checkStrptime("08/25/91", "%D", &tm2);
+-	checkStrptime("21.10.15", "%d.%m.%y", &tm3);
+-	checkStrptime("10.7.56 in 18th", "%d.%m.%y in %C th", &tm4);
++	UK_TEST_EXPECT_ZERO(checkStrptime("1991-08-25", "%Y-%m-%d", &tm2));
++	UK_TEST_EXPECT_ZERO(checkStrptime("25.08.91", "%d.%m.%y", &tm2));
++	UK_TEST_EXPECT_ZERO(checkStrptime("08/25/91", "%D", &tm2));
++	UK_TEST_EXPECT_ZERO(checkStrptime("21.10.15", "%d.%m.%y", &tm3));
++	UK_TEST_EXPECT_ZERO(checkStrptime("10.7.56 in 18th", "%d.%m.%y in %C th", &tm4));
+ 
+ 	/* Glibc */
+-	checkStrptime("1856-07-10", "%F", &tm4);
+-	checkStrptime("683078400", "%s", &tm2);
+-	checkStrptimeTz("+0200", 2, 0);
+-	checkStrptimeTz("-0530", -5, -30);
+-	checkStrptimeTz("-06", -6, 0);
+-
+-	return t_status;
++	UK_TEST_EXPECT_ZERO(checkStrptime("1856-07-10", "%F", &tm4));
```

Comment:
These tests fail and, judging by the comment, they might be applicable only for glibc.

Reply:
Yes, they are currently not working in `musl`. It is not a problem of Unikraft. @razvand Should they be commented? What do you think?

Reply:
@maniatro111 , I think they are OK at this point. At some point we would probably do a selection based on libc-type. But I wouldn't make it part of this release.

Having things working (i.e. all tests running, NOT passing) is fine. There shouldn't be crashes when running the tests. Other than that, it's fine.